### PR TITLE
Update documentation for forced colors mode

### PR DESCRIPTION
**Pull request by dlibby-**
Updates the list of properties that are forced.

Adds guidance around system color usage and the fact that they are respected in forced colors mode.

Explains that forced colors are chosen based on element, not role.

Adds an example of using a system color (previous example wouldn't have worked without `forced-color-adjust: none`).

### CODE REVIEW

**Comment by dlibby-** [comment]
@alisonmaher @melanierichards @minorninth, please take a first pass review (@ericwbailey feel free to hold off on review for now, but if you do want to review immediately, any feedback is welcome!)

**Comment by ericwbailey** [comment]
Thanks for the head's up, @dlibby-. If you don't mind @ mentioning me again when you'd like a review, I'd appreciate it.

**Review by alisonmaher — COMMENTED** [review]
This looks great, thanks for working on this @dlibby-.

Something that I noticed was that on the forced-color-adjust page, it suggests that Edge has shipped that with an alternate name (since we shipped with both). However, we don't mention -ms-high-contrast in the forced-colors page.

I'm not sure which is the right answer, but it would probably be good to keep them consistent.

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 1]
```diff
@@ -26,17 +22,19 @@ <h3 id="Values">Values</h3>
...
+ <dd>Indicates that forced colors mode is active. The browser provides the color palette to authors through the <a href="/en-US/docs/Web/CSS/color_value#System_Colors">CSS system color</a> keywords and, if appropriate, triggers the appropriate value of <code><a href="/en-US/docs/Web/CSS/@media/prefers-color-scheme">prefers-color-scheme</a></code> so that authors can adapt the page.</dd>
 </dl>
 
 <h2 id="Usage_notes">Usage notes</h2>
 
 <h3 id="Properties_affected_by_forced-color_mode">Properties affected by forced-color mode</h3>
 
-<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, user-specified values (if any) or browser-specified values are used instead.</p>
+<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, user-specified values (if any) or browser-specified values are used instead. These browser-specified values are selected from the set of system colors &emdash; this ensures a consistent contrast for common UI elements for users that have forced colors enabled.</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/@media/forced-colors/index.html:32]: It might be helpful to mention that the colors are forced at used value time, similar to the updated spec text: https://drafts.csswg.org/css-color-adjust-1/#forced-colors-properties.

Also, I don't believe that the second sentence regarding user-specified values applies anymore given that this is no longer implemented using revert.

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 2]
```diff
@@ -26,17 +22,19 @@ <h3 id="Values">Values</h3>
...
 
 <ul>
  <li>{{cssxref("color")}}</li>
+ <li>{{cssxref("background-color")}}</li>
```
Comment (alisonmaher) [inline files/en-us/web/css/@media/forced-colors/index.html:36]: It might make sense to move these to a separate section below, similar to the updated spec (along with the added scrollbar-color.)

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 3]
```diff
@@ -50,36 +48,44 @@ <h3 id="Properties_affected_by_forced-color_mode">Properties affected by forced-
...
+<p>The system colors that are forced for the above properties depend on the context of the element. For example the {{cssxref("color")}} property on button element will be forced to ButtonText, whereas on normal text it will be forced to CanvasText. See the list of system colors for additional details of when each might be appropriate in various UI contexts.</p>
+<p><strong>Note:</strong> user agents choose system colors based on elements, <em>not</em> their roles. As an example, adding role="button" to a div will not cause an element's color to be forced to ButtonText</p>
+
+<p>There are two cases where the user agent does not force the values for the above properties &emdash; when {{cssxref("forced-color-adjust")}} is applied to an element, or when a system color is specified by the author.</p>
+<p>When forced-color-adjust is specified on an element, none of the forced color values will apply, and author style will be applied as normal.</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/@media/forced-colors/index.html:55]: This should probably say something like "when forced-color-adjust is set to none" (same on the line above).

File: files/en-us/web/css/color_value/index.html [hunk 4]
```diff
@@ -926,19 +926,19 @@ <h3 id="System_Colors">System Colors</h3>
  </dd>
  <dt>ButtonFace</dt>
  <dd>
- <p>Background of push buttons</p>
+ <p>Background color of controls</p>
  </dd>
  <dt>ButtonText</dt>
  <dd>
- <p>Text of push buttons</p>
+ <p>Foreground or text color of controls</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/color_value/index.html:933]: To keep things consistent with the other system colors, perhaps this should be switched to "Text or foreground color", although just saying "Foreground color" might be simpler since I believe text should fall into that category.

**Review by alisonmaher — COMMENTED** [review]
Sorry, I'm not sure why these didn't post with my original comment.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 2 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:36]: Not sure if I got this right, based on the line you highlighted, but I did update to more closely match the spec's organization of the properties.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 3 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:55]: Indeed, thanks!

File: files/en-us/web/css/color_value/index.html (hunk 4 above)
Comment (dlibby-) [reply files/en-us/web/css/color_value/index.html:933]: Good point, I do think that is clearer.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 1 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:32]: I tried to put together a description of used value in layman's terms. Let me know what you think!

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 5]
```diff
@@ -26,60 +22,75 @@ <h3 id="Values">Values</h3>
...
 </dl>
 
 <h2 id="Usage_notes">Usage notes</h2>
 
 <h3 id="Properties_affected_by_forced-color_mode">Properties affected by forced-color mode</h3>
 
-<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, user-specified values (if any) or browser-specified values are used instead.</p>
+<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, browser-specified values are used instead. This does not affect the style cascade, but is accomplished by the browser using the browser-specified values instead of the cascaded values when painting colors. These browser-specified values are selected from the set of system colors &emdash; this ensures a consistent contrast for common UI elements for users that have forced colors enabled.</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/@media/forced-colors/index.html:32]: > This does not affect the style cascade, but is accomplished by the browser using the browser-specified values instead of the cascaded values when painting colors.

- The comma before "but" shouldn't be necessary.
- Up to you, but something along these lines might be slightly clearer: "The browser-specified values do not affect the style cascade; the values are instead forced by the browser at paint time."

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 6]
```diff
@@ -26,60 +22,75 @@ <h3 id="Values">Values</h3>
...
+ <li>{{cssxref("color-scheme")}} is forced to 'light dark'</li>
+ <li>{{cssxref("scrollbar-color")}} is forced to 'auto'</li>
+</ul>
+
+<p>The system colors that are forced for the above properties depend on the context of the element. For example the {{cssxref("color")}} property on button element will be forced to ButtonText, whereas on normal text it will be forced to CanvasText. See the list of system colors for additional details of when each might be appropriate in various UI contexts.</p>
+<p><strong>Note:</strong> user agents choose system colors based on elements, <em>not</em> their roles. As an example, adding role="button" to a div will not cause an element's color to be forced to ButtonText</p>
+
+<p>There are two cases where the user agent does not force the values for the above properties &emdash; when a {{cssxref("forced-color-adjust")}} value of 'none' is applied to an element, or when a system color is specified by the author.</p>
+<p>When forced-color-adjust is set to 'none' on an element, none of the forced color values will apply, and author style will be applied as normal.</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/@media/forced-colors/index.html:60]: "author styles" instead of "author style"

File: files/en-us/web/css/color_value/index.html [hunk 7]
```diff
@@ -926,19 +926,19 @@ <h3 id="System_Colors">System Colors</h3>
...
+ <p>Foreground color of controls</p>
  </dd>
  <dt>Canvas</dt>
  <dd>
  <p>Background of application content or documents</p>
  </dd>
  <dt>CanvasText</dt>
  <dd>
- <p>Text in application content or documents</p>
+ <p>Text or foreground color in application content or documents</p>
```
Comment (alisonmaher) [inline files/en-us/web/css/color_value/index.html:941]: Should the rest of these be updated to just "Foreground color", as well?

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 5 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html]: Yes, that is a good improvement, thanks!

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 6 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html]: Fixed.

File: files/en-us/web/css/color_value/index.html (hunk 7 above)
Comment (dlibby-) [reply files/en-us/web/css/color_value/index.html]: Yes, I that is a good clarification. Per discussion, left the link and mark related ones as specific to 'Text' since they are somewhat specialized.

**Review by alisonmaher — APPROVED** [review]

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 8]
```diff
@@ -26,60 +22,75 @@ <h3 id="Values">Values</h3>
  <dt><code><dfn>none</dfn></code></dt>
  <dd>Forced colors mode is not active; the page’s colors are not being forced into a limited palette.</dd>
  <dt><code>active</code></dt>
- <dd>Indicates that forced colors mode is active. The browser provides the color palette to authors through the <a href="/en-US/docs/Web/CSS/color_value#System_Colors">CSS system color</a> keywords and, if appropriate, it triggers the appropriate value of <code><a href="/en-US/docs/Web/CSS/@media/prefers-color-scheme">prefers-color-scheme</a></code> so that authors can adapt the page.</dd>
+ <dd>Indicates that forced colors mode is active. The browser provides the color palette to authors through the <a href="/en-US/docs/Web/CSS/color_value#System_Colors">CSS system color</a> keywords and, if appropriate, triggers the appropriate value of <code><a href="/en-US/docs/Web/CSS/@media/prefers-color-scheme">prefers-color-scheme</a></code> so that authors can adapt the page.</dd>
```
Comment (melanierichards) [inline files/en-us/web/css/@media/forced-colors/index.html:25]: Should we add a little bit here (or elsewhere in the doc) to describe when `prefers-color-scheme` gets triggered? At a minimum maybe link out to https://www.w3.org/TR/css-color-adjust-1/#forced?

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 9]
```diff
@@ -12,10 +12,6 @@
 
 <p><span class="seoSummary">The <strong><code>forced-colors</code></strong> <a href="/en-US/docs/Web/CSS">CSS</a> <a href="/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features">media feature</a> is used to detect if the {{Glossary("user agent")}} has enabled a forced colors mode where it enforces a user-chosen limited color palette on the page.</span></p>
```
Comment (melanierichards) [inline files/en-us/web/css/@media/forced-colors/index.html:13]: Should we update "a forced colors mode" to "a forced colors mode such as Windows High Contrast"? It's mentioned later on in the doc but might be good to lead with, for devs who have been trained to seek out guidance on HC specifically.

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 10]
```diff
@@ -26,60 +22,75 @@ <h3 id="Values">Values</h3>
...
+<ul>
+ <li>{{cssxref("box-shadow")}} is forced to 'none'</li>
+ <li>{{cssxref("text-shadow")}} is forced to 'none'</li>
+ <li>{{cssxref("background-image")}} is forced to 'none' for values that are not url-based</li>
+ <li>{{cssxref("color-scheme")}} is forced to 'light dark'</li>
+ <li>{{cssxref("scrollbar-color")}} is forced to 'auto'</li>
+</ul>
+
+<p>The system colors that are forced for the above properties depend on the context of the element. For example the {{cssxref("color")}} property on button element will be forced to ButtonText, whereas on normal text it will be forced to CanvasText. See the list of system colors for additional details of when each might be appropriate in various UI contexts.</p>
+<p><strong>Note:</strong> user agents choose system colors based on elements, <em>not</em> their roles. As an example, adding role="button" to a div will not cause an element's color to be forced to ButtonText</p>
```
Comment (melanierichards) [inline files/en-us/web/css/@media/forced-colors/index.html:57]:
```suggestion
<p><strong>Note:</strong> user agents choose system colors based on native element semantics, <em>not</em> on added ARIA roles. As an example, adding <code>role="button"</code> to a div will not cause an element's color to be forced to <code>ButtonText</code></p>
```

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 11]
```diff
@@ -26,60 +22,75 @@ <h3 id="Values">Values</h3>
...
+
+<p>There are two cases where the user agent does not force the values for the above properties &emdash; when a {{cssxref("forced-color-adjust")}} value of 'none' is applied to an element, or when a system color is specified by the author.</p>
+<p>When forced-color-adjust is set to 'none' on an element, none of the forced color values will apply, and author styles will be applied as normal.</p>
```
Comment (melanierichards) [inline files/en-us/web/css/@media/forced-colors/index.html:60]: We should mention that text backplates are removed as well.

Comment (melanierichards) [reply files/en-us/web/css/@media/forced-colors/index.html:60]: I guess we should add a generic section on that too :)

**Review by melanierichards — CHANGES_REQUESTED** [review]
Thanks @dlibby-! Added a couple minor suggestions.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 8 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:25]: Good point - added a sentence and link.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 10 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:57]: Thanks! I like that phrasing much better!

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 11 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html:60]: Added a paragraph on backplates and the fact that they are removed when adjustment is turned off.

**Review by melanierichards — APPROVED** [review]
LGTM, thanks for the updates!

**Comment by dlibby-** [comment]
@ericwbailey - ready for your review. PTAL when you get a chance, thanks!

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 12]
```diff
@@ -10,11 +10,7 @@
 ---
 <div>{{CSSRef}}</div>
 
-<p><span class="seoSummary">The <strong><code>forced-colors</code></strong> <a href="/en-US/docs/Web/CSS">CSS</a> <a href="/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features">media feature</a> is used to detect if the {{Glossary("user agent")}} has enabled a forced colors mode where it enforces a user-chosen limited color palette on the page.</span></p>
-
-<div class="notecard note">
-<p><strong>Note:</strong> This feature is not supported by any user agent and its specifics are subject to change.</p>
-</div>
+<p><span class="seoSummary">The <strong><code>forced-colors</code></strong> <a href="/en-US/docs/Web/CSS">CSS</a> <a href="/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features">media feature</a> is used to detect if the {{Glossary("user agent")}} has enabled a forced colors mode, such as Windows High Contrast, where it enforces a user-chosen limited color palette on the page.</span></p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:13]:
```suggestion
<p><span class="seoSummary">The <strong><code>forced-colors</code></strong> <a href="/en-US/docs/Web/CSS">CSS</a> <a href="/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features">media feature</a> is used to detect if the {{Glossary("user agent")}} has enabled a forced colors mode, where it enforces a user-chosen limited color palette on the page. An example of a forced colors mode is such as Windows High Contrast mode.</span></p>
```

Update this to simplify the sentence a bit.

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 13]
```diff
@@ -26,60 +22,81 @@ <h3 id="Values">Values</h3>
...
 </dl>
 
 <h2 id="Usage_notes">Usage notes</h2>
 
 <h3 id="Properties_affected_by_forced-color_mode">Properties affected by forced-color mode</h3>
 
-<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, user-specified values (if any) or browser-specified values are used instead.</p>
+<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, browser-specified values are used instead. The browser-specified values do not affect the style cascade; the values are instead forced by the browser at paint time. These browser-specified values are selected from the set of system colors &emdash; this ensures a consistent contrast for common UI elements for users that have forced colors enabled.</p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:32]:
```suggestion
<p>In forced colors mode, the values of the following properties are treated as if they have no author-level values specified. That is, browser-specified values are used instead. The browser-specified values do not affect the style cascade; the values are instead forced by the browser at paint time.</p>
<p>These browser-specified values are selected from the set of system colors &emdash; this ensures a consistent contrast for common UI elements for users that have forced colors enabled.</p>
```

Two thoughts, two paragraphs. This is kind of an abstract thing to grok, so I want to make it as simple to read as possible.

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 14]
```diff
@@ -26,60 +22,81 @@ <h3 id="Values">Values</h3>
...
+<p>Additionally the following properties have special behavior in forced colors mode:</p>
+<ul>
+ <li>{{cssxref("box-shadow")}} is forced to 'none'</li>
+ <li>{{cssxref("text-shadow")}} is forced to 'none'</li>
+ <li>{{cssxref("background-image")}} is forced to 'none' for values that are not url-based</li>
+ <li>{{cssxref("color-scheme")}} is forced to 'light dark'</li>
+ <li>{{cssxref("scrollbar-color")}} is forced to 'auto'</li>
+</ul>
+
+<p>The system colors that are forced for the above properties depend on the context of the element. For example the {{cssxref("color")}} property on button element will be forced to ButtonText, whereas on normal text it will be forced to CanvasText. See the list of system colors for additional details of when each might be appropriate in various UI contexts.</p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:56]:
```suggestion
<p>The system colors that are forced for the above properties depend on the context of the element. For example the {{cssxref("color")}} property on button element will be forced to <code>ButtonText</code>. On normal text it will be forced to <code>CanvasText</code>. See the list of system colors for additional details of when each might be appropriate in various UI contexts.</p>
```

Shortened a sentence here as well. In addition can we link to the list of system colors if we're naming it?

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 15]
```diff
@@ -26,60 +22,81 @@ <h3 id="Values">Values</h3>
...
+
+<p><strong>Note:</strong> user agents choose system colors based on native element semantics, <em>not</em> on added ARIA roles. As an example, adding <code>role="button"</code> to a div will not cause an element's color to be forced to <code>ButtonText</code></p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:58]:
```suggestion
<p><strong>Note:</strong> user agents choose system colors based on native element semantics, <em>not</em> on added ARIA roles. As an example, adding <code>role="button"</code> to a div will <strong>not</strong> cause an element's color to be forced to <code>ButtonText</code></p>
```

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 16]
```diff
@@ -26,60 +22,81 @@ <h3 id="Values">Values</h3>
...
+
+<p>In addition to these adjustments, browsers will help ensure text legibility by drawing “backplates” behind text. This is particularly important for preserving contrast when text is placed on top of images.
+</p>
+
+<p>There are two cases where the user agent does not force the values for the above properties &emdash; when a {{cssxref("forced-color-adjust")}} value of 'none' is applied to an element, or when a system color is specified by the author.</p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:63]:
```suggestion
<p>There are two cases where the user agent does not force the values for the above properties &emdash; when a {{cssxref("forced-color-adjust")}} value of <code>none</code> is applied to an element, or when a system color is specified by the author.</p>
```

File: files/en-us/web/css/@media/forced-colors/index.html [hunk 17]
```diff
@@ -26,60 +22,81 @@ <h3 id="Values">Values</h3>
...
+
+<p>When forced-color-adjust is set to 'none' on an element, none of the forced color values will apply, and author styles will be applied as normal. Additionally, the backplate for text will be disabled.</p>
```
Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:65]:
```suggestion
<p>When forced-color-adjust is set to <code>none</code> on an element, none of the forced color values will apply, and author styles will be applied as normal. Additionally, the backplate for text will be disabled.</p>
```

Comment (ericwbailey) [inline files/en-us/web/css/@media/forced-colors/index.html:65]: I'd like to see a sentence on the end of this paragraph that reinforces that this is something you should avoid manipulating at all costs, and only think about if the backplate is somehow obscuring legibility. 

I want to make it absolutely clear that disabling the backplate to art-direct a forced colors mode is not acceptable usage.

**Review by ericwbailey — CHANGES_REQUESTED** [review]
Good stuff. I left some suggestions, but from a high level I would like to see an accessibility concerns subsection that gives guidance that forced colors should only be tweaks if the content isn't legible, and not a reason to create a whole separate forced colors design.

I'd also like it to stress that this can be the final thing someone might be able to use in order to keep using their computer, so its appearance and method of working is highly intentional.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 12 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html]: Done.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 14 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html]: There was a previous link, but added another one here for convenience.

File: files/en-us/web/css/@media/forced-colors/index.html (hunk 17 above)
Comment (dlibby-) [reply files/en-us/web/css/@media/forced-colors/index.html]: Folded this into an overall Accessibility concerns sections, let me know if it is strong enough.

**Comment by dlibby-** [comment]
Thanks for the review! I think this is ready for another round.

**Review by bauerlumb — APPROVED** [review]

**Review by ericwbailey — APPROVED** [review]
Love it. Thank you so much for doing this.
